routemaster/client.rb

```diff
@@ -11,149 +14,171 @@
+        send_event('noop', topic, callback, timestamp, async: true)
+      end
 
-    def noop(topic, callback, timestamp = nil)
-      _send_event('noop', topic, callback, timestamp)
-    end
+      def subscribe(topics:, callback:, **options)
+        assert_valid_topics!(topics)
+        assert_valid_url!(callback)
+        assert_valid_max_events! options[:max] if options[:max]
+        assert_valid_timeout! options[:timeout] if options[:timeout]
 
-    def subscribe(options = {})
-      if (options.keys - [:topics, :callback, :timeout, :max, :uuid]).any?
-        raise ArgumentError.new('bad options')
+        conn.subscribe(topics: topics, callback: callback, **options)
       end
-      _assert options[:topics].kind_of?(Enumerable), 'topics required'
-      _assert options[:callback], 'callback required'
-      _assert_valid_timeout options[:timeout] if options[:timeout]
-      _assert_valid_max_events options[:max] if options[:max]
 
-      options[:topics].each { |t| _assert_valid_topic(t) }
-      _assert_valid_url(options[:callback])
-      _conn.subscribe(options)
-    end
+      def unsubscribe(*topics)
+        assert_valid_topics!(topics)
 
-    def unsubscribe(*topics)
-      topics.each { |t| _assert_valid_topic(t) }
+        topics.each do |t|
+          response = conn.delete("/subscriber/topics/#{t}")
 
-      topics.each do |t|
-        response = _conn.delete("/subscriber/topics/#{t}")
+          unless response.success?
+            raise 'unsubscribe rejected'
+          end
+        end
+      end
+
+      def unsubscribe_all
+        response = conn.delete('/subscriber')
 
         unless response.success?
-          raise 'unsubscribe rejected'
+          raise 'unsubscribe all rejected'
         end
       end
-    end
 
-    def unsubscribe_all
-      response = _conn.delete('/subscriber')
+      def delete_topic(topic)
+        assert_valid_topic!(topic)
+
+        response = conn.delete("/topics/#{topic}")
 
-      unless response.success?
-        raise 'unsubscribe all rejected'
+        unless response.success?
+          raise 'failed to delete topic'
+        end
       end
-    end
 
-    def delete_topic(topic)
-      _assert_valid_topic(topic)
+      def monitor_topics
+        response = conn.get('/topics') do |r|
+          r.headers['Content-Type'] = 'application/json'
+        end
 
-      response = _conn.delete("/topics/#{topic}")
+        unless response.success?
+          raise 'failed to connect to /topics'
+        end
 
-      unless response.success?
-        raise 'failed to delete topic'
+        Oj.load(response.body).map do |raw_topic|
+          Topic.new raw_topic
+        end
       end
-    end
 
-    def monitor_topics
-      response = _conn.get('/topics') do |r|
-        r.headers['Content-Type'] = 'application/json'
-      end
+      private
 
-      unless response.success?
-        raise 'failed to connect to /topics'
+      def conn
```

Comment:
I appreciate it's an uncommon style (for Ruby). I use it because it makes private method leakage obvious (both internally and from other classes).